### ISSUE TITLE
A question about the standardization of validation data sets

### ISSUE THREAD

**Issue by little1TouwaErio** [ChengBinJin/V-GAN-tensorflow#15]
Hello, I have been paying attention to your code for a long time. I haven't registered Github for a long time.There is something wrong with the code today, so I made a special registration to understand the problem.
In the preprocessing standardization process before the validation of the data set, fundus_imgs[index,..., 0] > 40.0
May I ask how this 40.0 was obtained? I don't quite understand and hope to get your help. Thank you very much.

**Comment by little1TouwaErio** [comment]
@ChengBinJin  I shall appreciate a reply at your earliest convenience. Thank you very much！

**Comment by ChengBinJin** [comment]
The fixed value 40.0 is from V-GAn author's Keras implementation. As my understanding, in the zero-centering preprocessing, we need to calculate the mean value of our training dataset. However, there is some background (black) in the retina images. If we calculate mean value, including all of the areas, the calculated mean value will smaller than the actual mean value due to the background's low intensities. Therefore, we have two options. One is that using mask images to extract foreground regions then calculate mean values. Another way is by using a threshold value to exclude background values and then calculate the mean value. In V-GAN implementation, we utilized the second more relaxed way. The value 40.0 is the threshold that we want to exclude all of the low-intensity background pixels. The author's experimental knowledge selects the value. Moreover, this threshold is also depended on the dataset you choose.

**Comment by little1TouwaErio** [comment]
I have understood this problem.Thank you for your prompt reply.

**Comment by Lililicg** [comment]
Thank you very much for your last rely. I have been modifying this code  to segment other medical images. But my understanding of a coefficient in this code is not very clear.
In the main.py file, a coefficient ratio_gan2seg is defined.  How is this constant coefficient obtained? Your prompt reply will be highly appreciate. @ChengBinJin

**Comment by little1TouwaErio** [comment]
Thank you very much for your last reply. I have been modifying this code  to segment other medical images. But my understanding of a coefficient in this code is not very clear.
In the main.py file, a coefficient ratio_gan2seg is defined. How is this constant coefficient obtained? Your prompt reply will be highly appreciate. @ChengBinJin

**Comment by ChengBinJin** [comment]
The `ratio_gan2seg` is 10 in this implementation. It means that the ratio between the GAN_loss and Seg_loss is 1:10. Obviously, Seg_loss is much more critical in the segmentation task. This ratio is prevalent in conditional GAN tasks such as pix2pix (1:100) and cycleGAN (1:10). If you set 1:1 between GAN_loss and Seg_loss, the predicted result is very realistic, but it's not correct compared to the gold standard. @Lililicg

**Comment by little1TouwaErio** [comment]
I am very grateful for your reply.
